http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -849,6 +851,34 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
         )
     } yield ()
 
+  private[monitor] def handleDeleteKubernetesClusterMessage(msg: DeleteKubernetesClusterMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      // TODO: ${ctx.traceId} ends up being None with manually published messages. Make sure that's not the case with cron-job-created messages
+      ctx <- ev.ask
+      clusterId = msg.clusterId
+      _ <- logger.info(
+        s"Beginning clean-up of cluster $clusterId in project ${msg.project} because it has had no apps running for a period of time. | trace id: ${ctx.traceId}"
+      )
+      // TODO: Should we check again that the cluster is okay to delete in case a user requested app creation since the cron job published the message?
+      _ <- kubernetesClusterQuery.markPendingDeletion(clusterId).transaction
+      _ <- gkeInterp
+      // TODO: Should we retry failures and with what RetryConfig? If all retries fail, send an alert?
+        .deleteAndPollCluster(DeleteClusterParams(msg.clusterId, msg.project))
+        .onError {
+          case _ =>
+            for {
+              _ <- logger.error(
+                s"An error occurred during clean-up of cluster ${clusterId} in project ${msg.project}. | trace id: ${ctx.traceId}"
+              )
+              _ <- kubernetesClusterQuery.updateStatus(clusterId, KubernetesClusterStatus.Error).transaction
+              // TODO: Create a KUBERNETES_CLUSTER_ERROR table to log the error message?
```

Comment:
Not sure we need a `KUBERNETES_CLUSTER_ERROR` table - this isn't really user facing. I wonder if this is a candidate for error reporting? I feel like we should be alerted if a cluster failed to be delete, which should have been deleted.